service/models.py

```diff
@@ -15,30 +15,34 @@
 # Function to initialize the database
 def init_db(app):
     """ Initializes the SQLAlchemy app """
-    YourResourceModel.init_db(app)
+    Shopcart.init_db(app)
+    Item.init_db(app)
 
 
 class DataValidationError(Exception):
     """ Used for an data validation errors when deserializing """
 
 
-class YourResourceModel(db.Model):
+class Shopcart(db.Model):
     """
-    Class that represents a YourResourceModel
+    Class that represents a Shopcart
     """
 
     app = None
 
     # Table Schema
-    id = db.Column(db.Integer, primary_key=True)
-    name = db.Column(db.String(63))
+    # customer_id (primary key), customer_name, shopcart_quantity, shopcart_price
+    customer_id = db.Column(db.Integer, primary_key=True)
+    customer_name = db.Column(db.String(63), nullable=False)
+    shopcart_quantity = db.Column(db.Integer, nullable=False, default=0)
+    shopcart_price = db.Column(db.Float, nullable=False, default=0.0)
 
```

Comment:
Yes, I think the two attributes are calculated from query either. What I think is that if adding an item in Item table then update the item_count and total_price in Shopcart table. So I set default value for the two attributes since there is no item_count and shopcart_price when adding a new shopcart. We can discuss this in detail.

Reply:
I see, thank you for your illustration 🙏🏻

Reply:
btw, I feel that item_count and total_price are more suitable to be calculated in the frontend; this way, we can eliminate at least one API call to update the shopcart table; also, if there's item price change, we wouldn't need to alter the shopcart table accordingly

Reply:
Yes they are supposed to be calculated in the frontend. But I'm not sure whether this will eliminate the API call to update the shopcart table. Can we discuss this in huddle when you are available?

Reply:
I think we are just developing backend services in this course?

Reply:
@LillisLuo My assumption is that we only need item_count and total_price in the frontend. If this is the case we might not need to store this data in the database? (this is how the API call to update the Shopcart table can be eliminated when adding an item).

Hope the clarification helps lol Or we can huddle later to discuss about this.

Reply:
@momo100110 it's said that we will also be developing frontend later in this semester

Reply:
Hi Kangle, yes we only need to develop the backend services. It is simple to make the calculations in the frontend. I think we can ask Professor to add the calculation function in the frontend maybe?

Reply:
@wl2484 Sorry I didn't get your point just now. Now I understand. You are correct we don't need to store the data in the database. We only need to make a GET call to acquire the quantities and price about items for one customer and let the frontend to calculate. Do you think we should delete the two attributes?

Reply:
If this also makes sense to you, I would vote for deleting the two attributes 😄

sorry again that I should have noticed this earlier when you proposed the design last week 🙈

Reply:
Don‘t worry about this, it is easy to operate since we just have a simple schema. Thanks for your suggestion, I will delete the two attributes.

Reply:
@wl2484 @LillisLuo  Hi Wan-Yu and Yuzhao, I found professor mentioned in another group that later we will add a UI but it will be a backend UI for the administrator of the website. According to this, I'm not sure whether we should do the calculation in the frontend.